src/Microsoft.AspNetCore.Mvc.Core/DependencyInjection/MvcCoreServiceCollectionExtensions.cs

```diff
@@ -266,6 +266,7 @@ internal static void AddMvcCoreServices(IServiceCollection services)
             services.TryAddEnumerable(
                 ServiceDescriptor.Singleton<EndpointDataSource, MvcEndpointDataSource>());
             services.TryAddSingleton<MvcEndpointInvokerFactory>();
+            services.TryAddSingleton<MvcEndpointFinder>();
```

Comment:
@rynowak DefaultLinkGenerator currently gets endpoint finder from DI, but since we always do TryAdd, we cannot be sure we can get Mvc's endpoint finder being used for link generation, so instead I added a property on LGContext which takes in endpoints. Routing's corresponding changes: https://github.com/aspnet/Routing/pull/574